core/domain/user_services.py

```diff
@@ -201,6 +210,42 @@ def get_users_settings(user_ids):
     return result
 
 
+def generate_profile_picture(user_id):
+    """Generates a profile picture for a new user."""
+    user_email = get_email_from_user_id(user_id)
+    user_gravatar = fetch_gravatar(user_email)
+    update_profile_picture_data_url(user_id, user_gravatar)
+
+
+def fetch_gravatar(email):
+    """Returns the gravatar corresponding to the user's email,
+    or an identicon generated from the email if the gravatar doesn't exist.
+
+    Note: if the call to the gravatar service fails,
+    this returns DEFAULT_IDENTICON_DATA_URL and logs an error.
+    """
+    gravatar_url = (
+        'http://www.gravatar.com/avatar/%s?d=identicon&s=%s' %
+        (hashlib.md5(email).hexdigest(), GRAVATAR_SIZE))
+    try:
+        result = urlfetch.fetch(
```

Comment:
@hitesh96db -- just to double-check, you've tested this, and the fetch has succeeded at least once, right (with an actual gravatar)?